images/Makefile

```diff
@@ -48,47 +84,57 @@ build-kernel: initrd-builder
 		distro=$$(basename $$(dirname $$kernel_dir)); \
 		image=$(REPOSITORY)/$$distro-kernel:$$version-$(ARCH); \
 		if [ "$(DRY_RUN)" = "true" ]; then \
-			echo "Dry run: Building kernel image: docker build -t $$image -f $$kernel_dir/Dockerfile.kernel $$kernel_dir"; \
+			echo "Dry run: Building kernel image: docker build -t $$image-$(TAG) -f $$kernel_dir/Dockerfile.kernel $$kernel_dir"; \
 		else \
-			echo "Building kernel image: $$image"; \
-			docker build -t $$image -f $$kernel_dir/Dockerfile.kernel $$kernel_dir; \
-		fi; \
-	done
-
-docker-push:
-	@for image in $$(docker images --format "{{.Repository}}:{{.Tag}}" | grep "$(REPOSITORY)" | grep "$(ARCH)"); do \
-		if [ "$(DRY_RUN)" = "true" ]; then \
-			echo "Dry run: Pushing image: docker push $$image"; \
-		else \
-			echo "Pushing image: $$image"; \
-			docker push $$image; \
+			echo "Building kernel image: $$image-$(TAG)"; \
+			docker build -t $$image-$(TAG) -f $$kernel_dir/Dockerfile.kernel $$kernel_dir || echo "FAIL: $$image-$(TAG)" >> failed.txt; \
+			if [ "$(PUSH)" = "true" ]; then \
+				echo "Pushing image: $$image-$(TAG)"; \
+				docker push $$image-$(TAG); \
+				if [ "$(LATEST)" = "true" ]; then \
+					echo "Pushing image: $$image-latest"; \
+					docker tag $$image-$(TAG) $$image-latest; \
+					docker push $$image-latest; \
+				fi; \
+			fi; \
+			docker image rm -f $$image-$(TAG); \
```

Comment:
TODO: move this "cleanup" behavior behind a new env variable so that it gets only enabled in CI.

Reply:
Done!